### ISSUE TITLE
Failing assertion when building with MKL and using Xception

### ISSUE THREAD

**Issue by abiro** [tensorflow/tensorflow#17047]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04.3 LTS
- **TensorFlow installed from (source or binary)**:
Source
- **TensorFlow version (use command below)**:
v1.6.0-rc0-19-gecec1d8
- **Python version**: 
Python 2.7.12
- **Bazel version (if compiling from source)**:
Bazel 0.8.0
- **GCC/Compiler version (if compiling from source)**:
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.5)
- **CUDA/cuDNN version**:
None
- **GPU model and memory**:
None
- **Exact command to reproduce**:
1. Get the Dockerfile from here: https://gist.github.com/abiro/c155a8107879d9c4e0368f3c3e94ea81
2. Build with the MKL library: `docker build -t tf_mkl -f Dockerfile.devel-cpu-mkl .`
3. Build without the MKL library: `docker build -t tf_nomkl -f Dockerfile.devel-cpu-mkl --build-arg MKL_FLAG=0 .`
3. Run image built with the MKL library: `docker run tf_mkl`
Result: error (see output below)
4. Run image built without the MKL library: `docker run tf_nomkl`
Result: no error

### Describe the problem
When building TensorFlow 1.6 with the MKL library, inference with the Xception model results in a failing assertion and the program exits. See the error message and the code below please. The Dockerfile linked above contains the source code and can be used to easily reproduce the problem.

#### Source code
```python
import tensorflow as tf
import numpy as np

x = tf.keras.applications.xception.Xception(weights="xception_weights.h5")
x.predict(np.zeros((1, 299, 299, 3)), batch_size=1)
```

#### Logs
Step 4 from above outputs the following:
>2018-02-15 15:04:26.610461: F tensorflow/core/kernels/mkl_input_conversion_op.cc:448] Check failed: tf_input.CheckReorderToOpMem( memory::primitive_desc(output_mkl_md, cpu_engine), tensor_out, &net) == true (0 vs. 1)
Aborted (core dumped)

#### Misc
The host machine used to build and run the Docker containers was an EC2 c4.2xlarge instance with an Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz processor.

Related to #16982

**Comment by bignamehyp** [comment]
@gunan can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @asimshankar: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @asimshankar: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by asimshankar** [comment]
@wei-v-wang : Is this related to the investigation in #16982 ?
(CC @tatianashp )

**Comment by wei-v-wang** [comment]
@asimshankar Sorry for the delayed answer.  It could be. I am working with my colleague to work on a fix for these. Hopefully the turnaround time will be better in short future.

**Comment by wei-v-wang** [comment]
@abiro  @asimshankar  @bignamehyp @gunan 

FYI: please try the latest commit e.g. https://github.com/tensorflow/tensorflow/commit/a175841eb549f069ac205fb32bf55314a387fe6d

I tested a private version similar to the above commit and did not encounter the issue anymore (but was indeed seeing this with some old public commits, sorry for the inconveniences). 

 investigate_17047]$ python p.py  
/usr/lib64/python2.7/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
2018-04-27 18:43:23.264129: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX512F
----
i.e. no error. 

@abiro could you please confirm? Thank you for reporting!

**Comment by wei-v-wang** [comment]
@abiro Thank you very much for following up and trying the latest commits and for reporting out the results. 
As we know TensorFlow has a large amount of commits each day. May I suggest using the tagged release versions, e.g. v1.7.0 tag and v1.8.0 tag. I have tested these tf commits, and both did not have the issues reported in this thread. The above "MklConv2DWithBiasBackpropBias" seems to be a new issue with latest code - we will investigate. But to unblock your work, may I suggest you try the **v180 (https://github.com/tensorflow/tensorflow/commit/93bc2e2072e0daccbcff7a90d397b704a9e8f778)** or **v170 (https://github.com/tensorflow/tensorflow/commit/024aecf414941e11eb643e29ceed3e1c47a115ad)**? 
Please see below for my results: 

[wangwei3@nervana-skx66 investigate_17047]$ python p.py   
/usr/lib64/python2.7/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
2018-05-08 13:07:20.992102: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2018-05-08 13:07:21.012545: I tensorflow/core/common_runtime/process_util.cc:63] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
[investigate_17047]$ cat p.py  
import tensorflow as tf
import numpy as np

x = tf.keras.applications.xception.Xception(weights="xception_weights.h5")
x.predict(np.zeros((1, 299, 299, 3)), batch_size=1)

[investigate_17047]$ ls xception_weights.h5  -hl
-rw-r--r-- 1 weiwang weiwang 88M May 24  2017 xception_weights.h5 


Or TF v170:

 investigate_17047]$ export PYTHONPATH=~/shared_big/TF_Public_v170                                                                                                                                                                               
[investigate_17047]$ python p.py                                                                                                                                                                                                                 
/usr/lib64/python2.7/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
2018-05-08 13:08:58.230349: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX512F

**Comment by abiro** [comment]
Thank you @wei-v-wang! I consider the issue resolved.
